cli/azd/cmd/util.go

```diff
@@ -431,13 +431,14 @@ func askOnePrompt(p survey.Prompt, response interface{}) error {
 }
 
 // promptTemplate ask the user to select a template.
-// An empty string is returned if the user selects 'Empty Template' from the choices
-func promptTemplate(ctx context.Context, message string, askOne Asker) (string, error) {
+// A template with empty values is returned if the user selects 'Empty Template' from the choices
+func promptTemplate(ctx context.Context, message string, askOne Asker) (templates.Template, error) {
 	templateManager := templates.NewTemplateManager()
 	templates, err := templateManager.ListTemplates()
+	result := templateManager.EmptyTemplate()
```

Comment:
FWIW - this doesn't feel super go'y to me.  The language enforces that `var result templates.Template` will be the zero value and `templates.Template{}` is another way to spell the zero value for an object.  I would expect that `EmptyTemplate` should only be called if there was some state we had to initialize on the object itself, which we don't need to do here.  Tn general gophers try to design APIs such that the zero value for types just works.

I'd just remove `EmptyTemplate()` and use one of the other patterns.

Reply:
I was having some issues trying to use `var result templates.Template`, but it was because there is another variable named `templates` on previous line 🤣   .  I've fixed it